packages/apollo-cli/src/commands/codegen/generate.ts

```diff
@@ -175,124 +175,65 @@ export default class Generate extends Command {
-    const pullFromEngine = !!apiKey && !flags.schema;
-
     const tasks: Listr = new Listr([
+      loadConfigStep(flags),
       {
-        title: "Scanning for GraphQL queries",
+        title: "Resolving GraphQL document sets and dependencies",
         task: async (ctx, task) => {
-          const paths = withGlobalFS(() => {
-            return (flags.queries ? flags.queries.split("\n") : []).flatMap(p =>
-              fg.sync(p)
-            );
-          });
-          task.title = `Scanning for GraphQL queries (${paths.length} found)`;
+          ctx.documentSets = await resolveDocumentSets(ctx.config, true);
 
-          const excludedPaths = [
-            flags.clientSchema ? path.resolve(flags.clientSchema) : undefined,
-            flags.schema ? path.resolve(flags.schema) : undefined
-          ];
-
-          ctx.queryPaths = paths.filter(p =>
-            !excludedPaths.some(v => v == path.resolve(p))
-          );
-        }
-      },
-      loadSchemaStep(
-        this,
-        pullFromEngine,
-        apiKey,
-        flags.engine,
-        "Loading GraphQL schema",
-        async ctx => {
-          if (flags.schema) {
-            ctx.schema = await fetchSchema({
-              endpoint: flags.schema
-            });
-          } else {
-            this.log("Not loading because no path was provided (you should have a client-side schema)");
-          }
-        }
-      ),
-      {
-        title: "Parsing GraphQL schema",
-        task: async (ctx, task) => {
-          if (ctx.schema) {
-            ctx.schema = buildClientSchema({ __schema: ctx.schema });
-          } else {
-            task.skip("No server-side schema provided")
-          }
-        }
-      },
-      {
-        title: "Loading client-side GraphQL schema",
-        task: async (ctx, task) => {
-          if (!flags.clientSchema) {
-            task.skip("Path to client schema not provided")
-          } else {
-            const foundDocuments = loadQueryDocuments([path.resolve(flags.clientSchema)]);
-            if (foundDocuments.length == 0) {
-              this.error("Found no query documents, aborting");
-            }
-
-            if (foundDocuments.length > 1) {
-              this.warn("Found more than one query document, using the first one");
-            }
-
-            const ast = foundDocuments[0];
-            visit(ast, {
-              enter(node) {
-                if (node.kind == "FieldDefinition") {
-                  (node as any).__client = true;
-                }
-              }
-            });
-
-            if (ctx.schema) {
-              ctx.schema = extendSchema(ctx.schema, ast);
-            } else {
-              ctx.schema = buildASTSchema(ast);
-            }
-          }
+          task.title = `Scanning for GraphQL queries (${(ctx.documentSets as ResolvedDocumentSet[])
+            .map(s => s.documentPaths.length)
+            .reduce((a, b) => a + b, 0)} found)`; // this is a bogus wrong value
```

Comment:
is this bogus because its files not operations?

Reply:
Oops! It used to be really bogus (counted the number of document sets) before. I'll drop the comment.